File: backend/internal/domain/entity/issue.go
package entity

import "time"

type Issue struct {
	ID          int       `json:"id"`
	ProjectID   int       `json:"project_id"`
	TypeID      uint      `json:"type_id"`
	PriorityID  uint      `json:"priority_id"`
	StatusID    uint      `json:"status_id"`
	Type        *Category `json:"type,omitempty"`
	Priority    *Category `json:"priority,omitempty"`
	Status      *Category `json:"status,omitempty"`
	Title       string    `json:"title"`
	Description *string   `json:"description"`
	AssigneeID  *int      `json:"assignee_id"`
	ReporterID  *int      `json:"reporter_id"`
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"`
}

File: frontend/src/lib/services/issue.ts
import { apiFetch, apiJson } from './api';

const API_URL = '/api/v1';

export interface Category {
    id: number;
    name: string;
    color: string;
    icon: string;
}

export interface Issue {
    id: number;
    project_id: number;
    type_id: number;
    priority_id: number;
    status_id: number;
    type?: Category;
    priority?: Category;
    status?: Category;
    title: string;
    description?: string | null;
    assignee_id?: number | null;
    reporter_id?: number | null;
    created_at: string;
    updated_at: string;
}

export type CreateIssuePayload = Omit<Issue, 'id' | 'created_at' | 'updated_at'>;

export const issueService = {
    async list(projectId: number, page = 1, limit = 10): Promise<{ items: Issue[], total: number }> {
        const res = await apiFetch(`${API_URL}/projects/${projectId}/issues?page=${page}&limit=${limit}`);
        if (!res.ok) throw new Error('Failed to load issues list');
        return res.json();
    },
    async create(projectId: number, payload: CreateIssuePayload): Promise<Issue> {
        return apiJson<Issue>(`${API_URL}/projects/${projectId}/issues`, {
            method: 'POST',
            body: JSON.stringify(payload)
        });
    },
    async update(projectId: number, issueId: number, payload: Partial<Issue>): Promise<Issue> {
        return apiJson<Issue>(`${API_URL}/projects/${projectId}/issues/${issueId}`, {
            method: 'PUT',
            body: JSON.stringify(payload)
        });
    },
    async delete(projectId: number, issueId: number): Promise<void> {
        await apiFetch(`${API_URL}/projects/${projectId}/issues/${issueId}`, { method: 'DELETE' });
    }
};
